include/libcrypto/bb_nn.h

```diff
@@ -3,22 +3,24 @@
 
 #include "ultratypes.h"
 
-typedef unsigned long element;
-typedef struct {
-    /* 0x0000 */ element e[8];
-} field_2n; /* size=0x20 */
+// TODO different header?
+typedef enum {
+    /* 0 */ BSL_TRUE,
+    /* 1 */ BSL_FALSE
+} BSL_boolean, boolean;
```

Comment:
What? I don't think I have ever seen this syntax for typedef before. What does it do? Typedef the same enum to those two typenames?

Reply:
Also, just noticed `TRUE` is zero and `FALSE` is one
![despair](https://user-images.githubusercontent.com/7416381/226779080-7bba0ee3-0091-479c-9254-9cde11796693.png)

Reply:
I noticed this too, but I assumed it was intentional and this was just a bizarre enum

Reply:
Yeah this enum is just all kinds of weird, we should re-evaluate whether we can get rid of it once we have full understanding of its usage.
The syntax there defines two types names for the same thing yeah. The libcrypto debug info shows that's what was done, again we should probably clean it up later.